src/palace/manager/api/odl/api.py

```diff
@@ -450,35 +400,48 @@ def _checkout(
                     raise NoAvailableCopies()
             raise
 
-        status = doc.get("status")
-
-        if status not in [self.READY_STATUS, self.ACTIVE_STATUS]:
+        if not doc.active:
             # Something went wrong with this loan and we don't actually
             # have the book checked out. This should never happen.
             # Remove the loan we created.
             _db.delete(loan)
             raise CannotLoan()
 
-        links = doc.get("links", [])
-        external_identifier = None
-        for link in links:
-            if link.get("rel") == "self":
-                external_identifier = link.get("href")
-                break
-        if not external_identifier:
+        # We save the link to the loan status document in the loan's external_identifier field, so
+        # we are able to retrieve it later.
+        loan_status_document_link: BaseLink | None = doc.links.get(
+            rel="self", type=LoanStatus.content_type()
+        )
+
+        # The ODL spec requires that a 'self' link be present in the links section of the response.
+        # See: https://drafts.opds.io/odl-1.0.html#54-interacting-with-a-checkout-link
+        # However, the open source LCP license status server does not provide this link, so we make
+        # an extra request to try to get the information we need from the 'status' link in the license
+        # document, which the LCP server does provide.
+        # TODO: Raise this issue with LCP server maintainers, and try to get a fix in place.
+        #   once that is done, we should be able to remove this fallback.
```

Comment:
Started a slack thread about this here: https://readium.slack.com/archives/C0SGTMMR6/p1727814143241779